### PR TITLE
Generalizing the UI to use QtPy

### PR DESCRIPTION
**Pull request by maljovec**
QtPy is agnostic of whether the user has PySide or PyQt installed, it will use whatever is available. In this way, we can utilize the current set of libraries + QtPy in order to run the interactive UIs.

--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

This is a follow-up on #161.

##### What are the significant changes in functionality due to this change request?

This changes the UI components to depend on QtPy rather than PySide. In this way, we can leverage whatever is available on the users machine and we don't need to force PySide if PyQt is already in use. For instance, conda will install PyQt as a prerequisite for matplotlib. The PySide that conda installs conflicts with the PyQt for the version of Qt to use, and thus a lower version number that fails on MacOS installs.  The problems of this lower version PySide can be circumvented with some advanced use of otool and install_name_change, but this is not guaranteed to work on a agnostic setup. Instead, if we install qtpy using conda, then we can leverage either PyQt or PySide seamlessly.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by maljovec** [comment]
Windows failure seems related to OS and not code failure:

```
Traceback (most recent call last):
  File "./backend_run_tests", line 128, in <module>
  File "C:/msys64/home/Civet/projects/raven/moose/python/TestHarness/TestHarness.py", line 34, in buildAndRun
    harness.findAndRunTests()
  File "C:/msys64/home/Civet/projects/raven/moose/python/TestHarness/TestHarness.py", line 192, in findAndRunTests
    self.launchTest(find_only, file, dirpath, testers)
  File "C:/msys64/home/Civet/projects/raven/moose/python/TestHarness/TestHarness.py", line 274, in launchTest
    self.runner.run(tester, command)
  File "C:/msys64/home/Civet/projects/raven/moose/python/TestHarness/RunParallel.py", line 115, in run
    tester.prepare(self.options)
  File "scripts/TestHarness/testers/RavenFramework.py", line 185, in prepare
    xmlopts = {}
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'C:/msys64/home/Civet/projects/raven/tests/framework/externalModel/testPrintHistorySet_dump_0.csv'
ERROR: exiting with code 1
```

**Comment by alfoa** [comment]
any news regarding this PR?

**Comment by maljovec** [comment]
It is awaiting approval. Development is complete.

**Comment by alfoa** [comment]
@maljovec did you talk with Paul Berg again to be sure?

**Comment by maljovec** [comment]
Not yet. I changed the status of this PR to WIP until I can verify with him.

File: framework/Application.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __init__(self, arg__1, messageHandler, interactionType=Interaction.Yes):
       self.printTag = 'RAVEN Application'
       self.UIs = {}
       self.interactionType = interactionType
-      qtg.QApplication.__init__(self,arg__1)
+      qtw.QApplication.__init__(self,arg__1)
```
Comment (PaulTalbot-INL) [inline framework/Application.py]: Unless there's a singular standard for "arg_1", the name should probably be changed to be RAVEN case.

File: framework/UI/ScatterView.py [hunk 2]
```diff
@@ -22,11 +22,12 @@
 warnings.simplefilter('default',DeprecationWarning)
 #End compatibility block for Python 3
 
+import qtpy
 import matplotlib
-matplotlib.rcParams['backend.qt4']='PySide'
+# matplotlib.rcParams['backend.qt4']=qtpy.API
```
Comment (PaulTalbot-INL) [inline framework/UI/ScatterView.py]: Do you want to keep this line?

File: framework/UI/ScatterView2D.py [hunk 3]
```diff
@@ -22,11 +22,13 @@
 warnings.simplefilter('default',DeprecationWarning)
 #End compatibility block for Python 3
 
+import qtpy
 import matplotlib
-matplotlib.rcParams['backend.qt4']='PySide'
+#matplotlib.rcParams['backend.qt4']=qtpy.API
```
Comment (PaulTalbot-INL) [inline framework/UI/ScatterView2D.py]: Do you want to keep this line?

File: framework/UI/ScatterView3D.py [hunk 4]
```diff
@@ -22,14 +22,16 @@
...
+from qtpy import QtCore as qtc
+from qtpy import QtGui as qtg
+from qtpy import QtWidgets as qtw
 
 from .BaseTopologicalView import BaseTopologicalView
 
 import matplotlib
 matplotlib.use('Qt4Agg')
-matplotlib.rcParams['backend.qt4']='PySide'
+#matplotlib.rcParams['backend.qt4']=qtpy.API
```
Comment (PaulTalbot-INL) [inline framework/UI/ScatterView3D.py]: Keep?

File: framework/UI/SensitivityView.py [hunk 5]
```diff
@@ -428,23 +429,23 @@ def layoutBarScene(self):
 
           if self.showLabelsAction.isChecked():
             txtItem = self.scene.addSimpleText(name,self.font)
-            # txtItem.setFlag(qtg.QGraphicsItem.ItemIgnoresTransformations)
+            # txtItem.setFlag(qtw.QGraphicsItem.ItemIgnoresTransformations)
```
Comment (PaulTalbot-INL) [inline framework/UI/SensitivityView.py]: Keep?

File: framework/UI/SensitivityView.py [hunk 6]
```diff
@@ -428,23 +429,23 @@ def layoutBarScene(self):
...
             txtItem.setPos(self.padding,y+0.5*(h-fontHeight))
-            txtItem.setFlag(qtg.QGraphicsItem.ItemIsMovable)
-            txtItem.setFlag(qtg.QGraphicsItem.ItemIsSelectable)
+            txtItem.setFlag(qtw.QGraphicsItem.ItemIsMovable)
+            txtItem.setFlag(qtw.QGraphicsItem.ItemIsSelectable)
             txtItem.setZValue(2)
           if self.showNumberAction.isChecked():
             numTxtItem = self.scene.addSimpleText('%.3g' % val, self.font)
-            # numTxtItem.setFlag(qtg.QGraphicsItem.ItemIgnoresTransformations)
+            # numTxtItem.setFlag(qtw.QGraphicsItem.ItemIgnoresTransformations)
```
Comment (PaulTalbot-INL) [inline framework/UI/SensitivityView.py]: Keep?

File: framework/UI/ZoomableGraphicsView.py [hunk 7]
```diff
@@ -307,7 +308,7 @@ def contextMenuEvent(self,event):
       right mouse button click.
       @ In, event, PySide.QtGui.QContextMenuEvent, the triggering event
     """
-    # if self.dragMode() == qtg.QGraphicsView.ScrollHandDrag:
+    # if self.dragMode() == qtw.QGraphicsView.ScrollHandDrag:
```
Comment (PaulTalbot-INL) [inline framework/UI/ZoomableGraphicsView.py]: Keep?

File: framework/UI/colors.py [hunk 8]
```diff
@@ -26,7 +26,7 @@
 from matplotlib import cm, colors
 import numpy as np
 import itertools
-from PySide.QtGui import QColor
+from qtpy.QtGui import QColor
 
 minPenColor = QColor(33,102,172) #QColor(57,94,150)
 minBrushColor = QColor(67,147,195) #QColor(114,143,184)
```
Comment (PaulTalbot-INL) [inline framework/UI/colors.py]: Keep, or at least comment on the color differences?

File: framework/Application.py (hunk 1 above)
Comment (maljovec) [inline framework/Application.py]: I was using the inherited argument name from [QApplication](https://srinikom.github.io/pyside-docs/PySide/QtGui/QApplication.html#PySide.QtGui.QApplication). I can change it to `arg1` in order to comply with RAVEN standards, so long as it does not lose its semantic meaning from the Qt side.

File: framework/UI/ScatterView.py (hunk 2 above)
Comment (maljovec) [inline framework/UI/ScatterView.py]: I think this can be safely removed.

File: framework/UI/ScatterView2D.py (hunk 3 above)
Comment (maljovec) [inline framework/UI/ScatterView2D.py]: Removing.

File: framework/UI/ScatterView3D.py (hunk 4 above)
Comment (maljovec) [inline framework/UI/ScatterView3D.py]: Removing.

File: framework/UI/SensitivityView.py (hunk 5 above)
Comment (maljovec) [inline framework/UI/SensitivityView.py]: Keep.

File: framework/UI/SensitivityView.py (hunk 6 above)
Comment (maljovec) [inline framework/UI/SensitivityView.py]: Keep.\

File: framework/UI/ZoomableGraphicsView.py (hunk 7 above)
Comment (maljovec) [inline framework/UI/ZoomableGraphicsView.py]: Keep.

File: framework/UI/colors.py (hunk 8 above)
Comment (maljovec) [inline framework/UI/colors.py]: removing commented colors.

**Comment by alfoa** [comment]
Well done!

**Pull request merged**